dali/kernels/imgproc/resample/resampling_impl_cpu.h

```diff
@@ -0,0 +1,272 @@
+  ));   // NOLINT
+}
+
+template <typename Out, typename In>
+inline void ResampleAxis(Surface2D<Out> out, Surface2D<In> in,
+                         const int *in_indices, const float *coeffs, int support, int axis) {
+  if (axis == 0)
+    ResampleVert(out, in, in_indices, coeffs, support);
+  else if (axis == 1)
+    ResampleHorz(out, in, in_indices, coeffs, support);
+  else
+    assert(!"Invalid axis index");
+}
+
+/// @brief Resamples `in` using Nearest Neighbor interpolation and stores result in `out`
+/// @param out - output surface
+/// @param in - input surface
+/// @param src_x0 - starting X coordinate of input
+/// @param src_y0 - starting Y coordinate of input
+/// @param scale_x - step of X input coordinate taken for each output pixel
+/// @param scale_y - step of Y input coordinate taken for each output row
+/// @remarks The function clamps input coordinates to fit in range defined by `in` dimensions.
+///          Scales can be negative to achieve flipping.
+template <typename Out, typename In>
+void ResampleNN(Surface2D<Out> out, Surface2D<const In> in,
+                float src_x0, float src_y0, float scale_x, float scale_y) {
+  assert(out.channels == in.channels);
+  assert(in.channel_stride == 1 && out.channel_stride == 1 ||
+         in.channels == 1 && out_channels == 1);
+  // assume HCW layout with contiguous pixels (not necessarily rows)
+  assert(out.pixel_stride == out.channels);
+
+  if (scale_x == 1) {
+    // FAST PATH - not scaling along X axis - just copy with repeated boundaries
+    int sx0 = std::floor(src_x0 + 0.5f);
+    int dx1 = sx0 + in.width;
+    if (dx1 > out.width)
+      dx1 = out.width;
+    int dx0 = 0;
+    if (sx0 < 0) {
+      dx0 = std::min(-sx0, out.width);
+    }
+
+    float sy = src_y0 + 0.5f * scale_y;
+    for (int y = 0; y < out.height; y++, sy += scale_y) {
+      int srcy = std::floor(sy);
+
+      if (srcy < 0) srcy = 0;
+      else if (srcy > in.height-1) srcy = in.height-1;
+
+      Out *out_ch = &out(0, y, 0);
+
+      int x;
+      const In *first_px = &in(0, srcy, 0);
+      for (x = 0; x < dx0; x++) {
+        for (int c = 0; c < out.channels; c++)
+          *out_ch++ = first_px[c];
+      }
+
+      const In *in_row = &in(sx0 + dx0, srcy, 0);
+      for (int j = x * out.channels; j < dx1 * out.channels; j++)
+          *out_ch++ = clamp<Out>(*in_row++);
+
+      x = dx1;
+      const In *last_px = &in(in.width-1, srcy, 0);
+      for (; x < out.width; x++) {
+        for (int c = 0; c < out.channels; c++)
+          *out_ch++ = last_px[c];
+      }
+    }
+    return;
+  }
+
+  constexpr int max_span_width = 256;
+  int col_offsets[max_span_width];  // NOLINT (kOnstant)
+
+  for (int x0 = 0; x0 < out.width; x0 += max_span_width) {
+    float sy = src_y0 + 0.5f * scale_y;
+
+    int span_width = x0 + max_span_width <= out.width ? max_span_width : out.width - x0;
+
+    for (int j = 0; j < span_width; j++) {
+      int x = x0 + j;
```

Comment:
```suggestion
      const int x = x0 + j;
```

Reply:
I think that in such a short function adding words just creates more visual clutter and doesn't solve any problems.

Reply:
The short's scope but those variables are quite confusable: `x x0 sx srcx src_x0`

Reply:
I'm nitpicking and it might be considered personal opinion, but it would help me reading your code :+1: